src/plugins/data/server/search/routes/msearch.ts

```diff
@@ -67,16 +67,23 @@ export function registerMsearchRoute(router: IRouter, deps: SearchRouteDependenc
             })
           ),
         }),
+        query: schema.object({
```

Comment:
could we actually just shove the data source id into each of the searches.body?  and parse out that value to get the client?